third_party/opa/data-and-policies/data_policies/action_struct.rego

```diff
@@ -0,0 +1,124 @@
+package data_policies
+
+#general enforcment action structure
+enforcement_action_struct = {
+    "action_name" : "<name of action>",
+    "desription" : "<free text description of the action>",
+    "arguments" : "<arguments set can be different for each action>",
+    "used_policy" : "<used_policy_struct>"
+}
+
+build_action_from_policies(used_policy) = action {
+    action = {
+        "used_policy" : used_policy
+    }
+}
+
+build_action_from_name(action_name, used_policy) = action {
+    action = {
+        "action_name" : action_name,
+        "desription" : action_name,
+        "arguments" :[],
+        "used_policy" : used_policy
+    }
+}
+
+build_action(action_name, description, arguments, used_policy) = action {
+    action = {
+        "action_name" : action_name,
+        "description" : description,
+        "arguments" :arguments,
+        "used_policy" : used_policy
+    }
+}
+
+################################### Enforcement Actions #######################################
+
+#deny access
+deny_access_struct = {
+    "action_name" : "deny access",
+    "description" : "Access to this data asset is denied",
+    "arguments" : {},
+    "used_policy" : "<used_policy_struct>"
+}
+
+build_deny_access_action(used_policies) = action {
```

Comment:
Wouldn't it be better to create generalized functions?  As per my previous comment, I think the actions should be in a taxonomy and not hardcoded.  I don't think we want to have to rebuild and redeploy if a new action is created or an existing one is changed.  Right?